Richer metadata

gadomski opened the pull request:
**Related Issue(s):** None

**Description:** Add more collection links and include the raster extension. Also add pre-commit.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] `examples/` is updated to reflect new STAC objects, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

pjhartzell reviewed (commented):
**Collection**
- There are newlines (\n\n) in the Collection description
- Add the Raster Extension to the collection STAC extension list
- Did you mean to leave placeholders for Month and Year in sci:citation?
- Keywords. Style question. Is all lowercase preferable?

**Items**
- Missing Raster Extension in hi_1992_mosaic_ccap_land_cover.json. Intentional? Also missing in other example Items.

gadomski commented:
> There are newlines (\n\n) in the Collection description

Fixed.

> Add the Raster Extension to the collection STAC extension list

Done.

> Did you mean to leave placeholders for Month and Year in sci:citation?

Nope, fixed.

> Keywords. Style question. Is all lowercase preferable?

I _think_ so, per https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#searchable-identifiers.

> Missing Raster Extension in hi_1992_mosaic_ccap_land_cover.json. Intentional? Also missing in other example Items.
Really? I see it here: https://github.com/stactools-packages/noaa-c-cap/pull/13/files#diff-5966faf116c2b920dbae49d6b653254c61d254f029914be928736ee6ff3cfdcfR224.

pjhartzell commented:
Regarding keywords being lowercase, agreed on best practices. Just may need to change in ETL template (not sure if that converts to capitalized words by default)

Yep, the raster extensions are in there. Not sure what I was looking at.
